trg-checks-dashboard/internal/templating/tractusx.go

```diff
@@ -0,0 +1,185 @@
+
+package templating
+
+import (
+	"context"
+	"log"
+	"os"
+	"sort"
+	"strings"
+
+	"github.com/eclipse-tractusx/tractusx-quality-checks/pkg/container"
+	"github.com/eclipse-tractusx/tractusx-quality-checks/pkg/docs"
+	"github.com/eclipse-tractusx/tractusx-quality-checks/pkg/helm"
+	"github.com/eclipse-tractusx/tractusx-quality-checks/pkg/repo"
+	"github.com/eclipse-tractusx/tractusx-quality-checks/pkg/tractusx"
+	"github.com/google/go-github/v53/github"
+	"golang.org/x/oauth2"
+)
+
+const gitHubOrg = "eclipse-tractusx"
+
+var gitHubClient *github.Client
+
+func CheckProducts() ([]CheckedProduct, []Repository) {
+	repoInfoByRepoUrl := make(map[string]repoInfo)
+	var unhandledRepos []Repository
+
+	repos := getOrgRepos()
+
+	for _, repo := range repos {
+		metadata := getMetadataForRepo(repo)
+
+		if metadata == nil {
+			unhandledRepos = append(unhandledRepos, Repository{Name: repo.Name, URL: repo.Url})
+		} else {
+			repoInfoByRepoUrl[repo.Url] = repoInfo{metadata: *metadata, repoName: repo.Name, repoUrl: repo.Url}
+		}
+	}
+
+	var checkedProducts []CheckedProduct
+	for _, p := range getProductsFromMetadata(repoInfoByRepoUrl) {
+		checkedProduct := CheckedProduct{Name: p.Name, LeadingRepo: p.LeadingRepo, OverallPassed: true}
+		for _, r := range p.Repositories {
+			checkedRepo := runQualityChecks(r)
+			checkedProduct.OverallPassed = checkedProduct.OverallPassed && checkedRepo.PassedAllGuidelines
+			checkedProduct.CheckedRepositories = append(checkedProduct.CheckedRepositories, checkedRepo)
+		}
+
+		checkedProducts = append(checkedProducts, checkedProduct)
+	}
+
+	return checkedProducts, unhandledRepos
+}
+
+func runQualityChecks(repo Repository) CheckedRepository {
+	log.Printf("Starting checks for repo: %s", repo.Name)
+	checkedRepo := CheckedRepository{RepoUrl: repo.URL, RepoName: repo.Name, PassedAllGuidelines: true}
+
+	dir, err := cloneRepo(repo)
+	if err != nil {
+		log.Printf("Could not clone repo %s. Error: %s", repo.URL, err)
+		return CheckedRepository{}
+	}
+
+	for _, check := range initializeChecksForDirectory(dir) {
+		testResult := check.Test()
+		checkedRepo.PassedAllGuidelines = checkedRepo.PassedAllGuidelines && (testResult.Passed || check.IsOptional())
+
+		guidelineCheck := GuidelineCheck{
+			Passed:           testResult.Passed,
+			Optional:         check.IsOptional(),
+			ErrorDescription: testResult.ErrorDescription,
+			GuidelineUrl:     check.ExternalDescription(),
+			GuidelineName:    check.Name(),
+		}
+		checkedRepo.GuidelineChecks = append(checkedRepo.GuidelineChecks, guidelineCheck)
+	}
+
+	return checkedRepo
+}
+
+func initializeChecksForDirectory(dir string) []tractusx.QualityGuideline {
+	var checks []tractusx.QualityGuideline
+
+	checks = append(checks, docs.NewReadmeExists(dir))
+	checks = append(checks, docs.NewInstallExists(dir))
+	checks = append(checks, docs.NewChangelogExists(dir))
+	//checks = append(checks, repo.NewRepoStructureExists(dir))
```

Comment:
Not sure why we may not want this check to happen, any reason why we repo structure check commented out?

Reply:
Oh I actually forgot to mentioned it in the PR description. There is an issue that I didn't see when adding the "baseDir" addition. eclipse-tractusx/tractusx-quality-checks#42 -> the call stack uses a function, that is not able to take a baseDir and fails. Therefor the whole check fails, when running on a different dir than the current one